lib/events/dynamoevents/dynamoevents.go

```diff
@@ -1313,7 +1351,7 @@ func (l *Log) migrateMatchingEvents(ctx context.Context, filterExpr string, tran
 			writeRequests = writeRequests[top:]
 
 			// Don't exceed maximum workers.
-			for workerCounter.Load() >= maxMigrationWorkers {
+			for workerCounter.Load() >= int32(migrationWorkers) {
```

Comment:
At this point in code, the int32 cast may look unwarranted and potentially unsafe. Could `approximateOptimalMigrationWorkers` be adapted to return int32 since it's over there the restriction to values less than `maxMigrationWorkers` takes place, and so the function's type can be made to reflect the fact int32 is safe to hold the value?

Reply:
Fixed, that makes sense for readability